kafka/src/test/java/org/apache/apex/malhar/kafka/KafkaInputOperatorTest.java

```diff
@@ -232,33 +228,11 @@ public void endWindow()
       int tupleSize = windowTupleCollector.size();
       tupleCollection.addAll(windowTupleCollector);
       
-      int countDownTupleSize = countDownAll ? tupleSize : endTuples;
+      int countDownTupleSize = endTuples;
 
       if (latch != null) {
-        Assert.assertTrue("received END_TUPLES more than expected.", latch.getCount() >= countDownTupleSize);
-        while (countDownTupleSize > 0) {
+        while (countDownTupleSize-- > 0) {
             latch.countDown();
-            --countDownTupleSize;
-        }
-        if (latch.getCount() == 0) {
-          /**
-           * The time before countDown() and the shutdown() of the application
```

Comment:
shutdown() set appDone = true; it is different from alive. I am not sure wether the following steps will set alive to false or not. Even it will be set, it still could be later than the checking as the Thread check appDone will sleep 1 second after 1 loop.
alive could be set when handling ShutdownException or teardown()